tst/unit/test_error_checking.cpp

```diff
@@ -0,0 +1,40 @@
+//========================================================================================
+// Parthenon performance portable AMR framework
+// Copyright(C) 2020 The Parthenon collaboration
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S. Department of Energy/National Nuclear
+// Security Administration. All rights in the program are reserved by Triad
+// National Security, LLC, and the U.S. Department of Energy/National Nuclear
+// Security Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide license
+// in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do
+// so.
+//========================================================================================
+
+#include <catch2/catch.hpp>
+
+#include "utils/error_checking.hpp"
+
+TEST_CASE("Parthenon Error Checking", "[ErrorChecking][Kokkos]") {
+  SECTION("PARTHENON_REQUIRE passes if condition true") {
+    PARTHENON_REQUIRE(true, "This shouldn't fail");
+  }
+#ifdef PARTHENON_TEST_ERROR_CHECKING
+  SECTION("PARTHENON_REQUIRE fails if condition false") {
+    PARTHENON_REQUIRE(false, "This should fail");
```

Comment:
Do you not need to comment out each of these sections to check the subsequent one?

Reply:
You do. I don't know a way around that unfortunately. 

One possibility I guess is I could give them all different tags and then run them or not via `ctest` rather than using the `ifdef`. Not sure which is better.

Reply:
idk, unless you expect people are going to run this on occassion, I'd say you should just delete this code.

What you could potentially do is just create a non-catch2 executable that causes the failure, and then manually call `add_test`, followed by setting this test property: https://cmake.org/cmake/help/latest/prop_test/WILL_FAIL.html#prop_test:WILL_FAIL

Reply:
> I'd say you should just delete this code.

I wanted it in as examples of use. A bit of documentation. I'll delete it if you think I should though.

Reply:
I'll leave it up to you.

Reply:
:+1: I'll keep it then.